src/context.ts

```diff
@@ -76,7 +76,8 @@ export class PageContext {
     debug.pages('add', path)
     for (const p of toArray(path)) {
       const pageDirPath = slash(resolve(this.root, pageDir.dir))
-      const route = slash(join(pageDir.baseRoute, p.replace(`${pageDirPath}/`, '').replace(extname(p), '')))
+      const extension = ensurePrefix('.', this.options.extensions.find(e => p.endsWith(e)) ?? extname(p))
```

Comment:
`extname()` will return extname with `.` prefix so we don't need to wrap it with `ensurePrefix`.

Reply:
Yes, at first I wrote the following but changed the code because `ensurePrefix` does no harm even if the prefix already exists.

```javascript
let extension = this.options.extensions.find(e => p.endsWith(e))
if (extension) {
  extension = ensurePrefix(".", extension)
} else {
  extension = extname(p)
}
```

In the first place, should we fallback to `extname` function when the file does not match any extension? When users specify `extensions` option explicitly, they do not expect files with any other extension to be converted to pages, do they?

Reply:
> In the first place, should we fallback to extname function when the file does not match any extension?

Yes, I think remove extname fallback make sense to me.